deployment-tinkerer/src/main/java/org/wso2/testgrid/deployment/tinkerer/utils/SSHHelper.java

```diff
@@ -0,0 +1,185 @@
+                            "  " +
+                            "ProxyCommand ssh %s@%s -W %%h:%%p%n"
+                    , hostEntry, agent.getInstanceUser(), agent.getInstanceIp(),
+                    keyPath.toAbsolutePath().toString(), agent.getInstanceUser(), bastianEntry);
+            Files.write(config, entry.getBytes(StandardCharsets.UTF_8), StandardOpenOption.APPEND);
+        }
+    }
+
+    /**
+     * Adds the config entry when there is no bastian instance.
+     * This method can be used the instance can be accessed directly
+     *
+     * @param keyContent Contents of the security key file used to access the instance
+     * @param agent The {@link Agent} object corresponding to the remote agent running on the instance
+     * @throws IOException when there is an error crating the config entry
+     */
+    public static void addConfigEntry(String keyContent, Agent agent) throws IOException {
+
+        Path config = Paths.get(System.getProperty("user.home"), ".ssh", "config");
+        Path keyPath = Paths.get(System.getProperty("user.home"), ".ssh", agent.getInstanceId() + "-key.pem");
+        saveKeyFile(keyContent, keyPath);
+        File file = config.toFile();
+        //Create the file if it does not exist
+        if (!Files.exists(config) || !Files.isRegularFile(config)) {
+            boolean mkdirs = file.getParentFile().mkdirs();
+            boolean newFile = file.createNewFile();
+            if (mkdirs || newFile) {
+                logger.info("Created new ssh config file");
+            }
+        }
+
+        String hostEntry = "host:" + agent.getInstanceName();
+        if (!containsEntry(file, agent.getInstanceId())) {
+            //add Host entry
+            String entry = String.format("Host %s%n  StrictHostKeyChecking=no%n" +
+                            "  UserKnownHostsFile=/dev/null%n" +
+                            "  User %s%n" +
+                            "  HostName %s%n" +
+                            "  IdentityFile %s%n" +
+                            "  "
+                    , hostEntry, agent.getInstanceUser(), agent.getInstanceIp(),
+                    keyPath.toAbsolutePath().toString());
+            Files.write(config, entry.getBytes(StandardCharsets.UTF_8), StandardOpenOption.APPEND);
+        }
+    }
+
+    /**
+     * Checks if a particular host entry has already been made in the config file
+     * @param file File object for the ssh config file
+     * @param entry Entry to be checked
+     * @return true if the entry is already existing , false otherwise
+     * @throws FileNotFoundException when there is an error finding the config file
+     */
+    private static boolean containsEntry(File file, String entry) throws FileNotFoundException {
+        Scanner scanner = new Scanner(file, StandardCharsets.UTF_8.name());
+        while (scanner.hasNextLine()) {
+            String line = scanner.nextLine();
+            if (line.contains(entry)) {
+                return true;
```

Comment:
This is good. At the same time, we also have a Java 8 streaming API to do the same in a concise way. :)